### ISSUE TITLE
Initial support for Local Zones

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#11136]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

[AWS Local Zones](https://aws.amazon.com/about-aws/global-infrastructure/localzones/) are now available.
Add initial support to the AZ data sources.

### New or Affected Resource(s)

* [`aws_availability_zone`](https://www.terraform.io/docs/providers/aws/d/availability_zone.html)
* [`aws_availability_zones`](https://www.terraform.io/docs/providers/aws/d/availability_zones.html)
### Potential Terraform Configuration

```hcl
data "aws_availability_zones" "available_local" {
  state = "available"

  filter {
    name   = "opt-in-status"
    values = ["opted-in"]
  }
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2019/12/introducing-aws-local-zone-in-los-angeles-ca/).
[Blog post](https://aws.amazon.com/blogs/aws/aws-now-available-from-a-local-zone-in-los-angeles/).

Requires [AWS SDK v1.25.46](https://github.com/aws/aws-sdk-go/releases/tag/v1.25.46):
* https://github.com/terraform-providers/terraform-provider-aws/pull/11116

**Comment by ewbankkit** [comment]
Related:
* https://github.com/terraform-providers/terraform-provider-aws/issues/5835

**Comment by ewbankkit** [comment]
AWS Local Zones can now be enabled programmatically  - [announcement](https://aws.amazon.com/about-aws/whats-new/2020/03/aws-customers-enable-aws-local-zones-themselves/).

Requires [AWS SDK v1.29.19](https://github.com/aws/aws-sdk-go/releases/tag/v1.29.19):
* https://github.com/terraform-providers/terraform-provider-aws/pull/12294

**Comment by bflad** [comment]
Pull request submitted to cover all the changes here: https://github.com/terraform-providers/terraform-provider-aws/pull/12400

**Comment by bflad** [comment]
Support for the following has been merged and will release with version 2.55.0 of the Terraform AWS Provider, later this week.

FEATURES:

* **New Resource:** `aws_ec2_availability_zone_group`

ENHANCEMENTS:

* data-source/aws_availability_zone: Add `all_availability_zones` and `filter` arguments
* data-source/aws_availability_zone: Add `group_name`, `network_border_group`, and `opt_in_status` attributes
* data-source/aws_availability_zones: Add `all_availability_zones` and `filter` arguments
* data-source/aws_availability_zones: Add `group_names` attribute
